### ISSUE TITLE
Icalendar not working for dates in january

### ISSUE THREAD

**Issue by Patanouque** [n8n-io/n8n#2627]
**Describe the bug**

When I use the icalendar block for a date other than january  it works, but the exact same block for dates in january does not work anymore

**To Reproduce**

Simply use a ical block and set start date and/or end date in january

This works :
```
{
  "nodes": [
    {
      "parameters": {},
      "name": "Start",
      "type": "n8n-nodes-base.start",
      "typeVersion": 1,
      "position": [
        240,
        300
      ]
    },
    {
      "parameters": {
        "title": "blabla",
        "start": "2021-12-07T15:17:54.000Z",
        "end": "2021-12-15T15:17:50.000Z",
        "additionalFields": {}
      },
      "name": "iCalendar",
      "type": "n8n-nodes-base.iCal",
      "typeVersion": 1,
      "position": [
        520,
        300
      ],
      "alwaysOutputData": false
    }
  ],
  "connections": {
    "Start": {
      "main": [
        [
          {
            "node": "iCalendar",
            "type": "main",
            "index": 0
          }
        ]
      ]
    }
  }
}
```
![image](https://user-images.githubusercontent.com/27449850/147948627-9260d4c5-a432-4616-8555-3453ccfd85fd.png)

This does not work
```
{
  "nodes": [
    {
      "parameters": {},
      "name": "Start",
      "type": "n8n-nodes-base.start",
      "typeVersion": 1,
      "position": [
        240,
        300
      ]
    },
    {
      "parameters": {
        "title": "blabla",
        "start": "2022-01-03T15:17:54.000Z",
        "end": "2022-01-04T15:17:50.000Z",
        "additionalFields": {}
      },
      "name": "iCalendar",
      "type": "n8n-nodes-base.iCal",
      "typeVersion": 1,
      "position": [
        520,
        300
      ],
      "alwaysOutputData": false
    }
  ],
  "connections": {
    "Start": {
      "main": [
        [
          {
            "node": "iCalendar",
            "type": "main",
            "index": 0
          }
        ]
      ]
    }
  }
}
```
![image](https://user-images.githubusercontent.com/27449850/147948462-b996b997-d145-48f7-9b2e-cec7c99dfec0.png)

Also I tried with only one date (start or finish) in january and I get this error

![image](https://user-images.githubusercontent.com/27449850/147949280-1bb0dd80-7178-4a46-ae36-546ee71ff882.png)

It seems like there is a check that fails for january

**Comment by Joffcom** [comment]
Hey @Patanouque,

I just took a quick look and I was able to reproduce the issue, I have made a note to have a proper look tomorrow. Thanks for reporting this :+1:

**Comment by Joffcom** [comment]
Hey @Patanouque,

I have just put in a pull request to resolve this issue, As a temporary work around what you could do is use the Date & Time node to set the date and increase the month by one so for January use February.

**Comment by Patanouque** [comment]
Hello, thanks a lot
I had not even noticed but actually all the dates were offset by 1 month so your fix works great 

Thanks for your fix and for n8n that is a joy to use !!

**Comment by janober** [comment]
Fix got released with n8n@0.158.0
